sympy/functions/elementary/trigonometric.py

```diff
@@ -2221,6 +2221,8 @@ class asec(InverseTrigonometricFunction):
 
     @classmethod
     def eval(cls, arg):
+        if arg.is_zero:
+            return S.NaN
```

Comment:
``asec`` is not defined at 0, and I have not been able to find a reference to its limit in the docs.
As ``acos(x) = asec(1/x)``, the limit should be the same as that of ``acos`` at ``zoo``. That is, in fact, given by [wolfram](http://functions.wolfram.com/ElementaryFunctions/ArcCos/03/02/) as ``zoo``. 
Currently, SymPy leaves ``acos(zoo)`` unevaluated but the closely related limit ``acosh(zoo)`` is given as ``oo``. So, according to wolfram, these should be ``zoo``, too.